README.md

```diff
@@ -28,24 +29,43 @@ header.
 ```rust
 let token = encode(&Header::default(), &my_claims, "secret".as_ref()).unwrap();
 ```
-In that example, `my_claims` is an instance of a Claims struct that derives `RustcEncodable` and `RustcDecodable`.
+In that example, `my_claims` is an instance of a Claims struct that derives `Serialize` and `Deserialize`.
 The default algorithm is HS256.
 Look at custom headers section to see how to change that.
 
 ### Decoding
 ```rust
-let token = decode::<Claims>(&token, "secret", Algorithm::HS256).unwrap();
+let token = decode::<Claims>(&token, "secret", Algorithm::HS256, &Validation::default()).unwrap();
 // token is a struct with 2 params: header and claims
 ```
-In addition to the normal base64/json decoding errors, `decode` can return two custom errors:
+`decode` can error for a variety of reasons:
 
-- **InvalidToken**: if the token is not a valid JWT
-- **InvalidSignature**: if the signature doesn't match
-- **WrongAlgorithmHeader**: if the alg in the header doesn't match the one given to decode
+- the token or its signature is invalid
+- error while decoding base64 or the result of decoding base64 is not valid UTF-8
+- validation failed
```

Comment:
I wonder if this could be slightly more verbose with something like `validation of at least one reserved claim failed`. `reserved claim` is a bit awkward, but that's what the spec refers to them as.